### PR TITLE
output dir checking and schema

### PR DESCRIPTION
**Pull request by stolarczyk**
- add output directory checking and creation (in case it does not exist)
- add input PEP schema
- other minor enhancements

### CODE REVIEW

File: bed_maker.py [hunk 1]
```diff
@@ -113,22 +103,19 @@ def main():
         print("Determined path to chrom.sizes asset: {}".format(chrom_sizes))
         # define a target for temporary bw files
         temp_target = os.path.join(sample_folder, file_id + ".bw")
+        pm.clean_add(temp_target)
```
Comment (joseverdezoto) [inline bed_maker.py:108]: @michalstolarczyk shouldn't we clean the intermediate files directory after producing the bed files (running `cmd`)?

**Review by joseverdezoto — COMMENTED** [review]
Intermediate files directory cleaning

File: bed_maker.py (hunk 1 above)
Comment (stolarczyk) [reply bed_maker.py:108]: the intermediate files can be added to the cleanup list anytime and will be cleaned upon successful pipeline run. This guide explains this concept well: https://pypiper.readthedocs.io/en/latest/clean/#cleaning-up-intermediate-files

Comment (joseverdezoto) [reply bed_maker.py:108]: Right, I forgot that the intermediate files are cleaned upon pipeline completion.

**Pull request merged**